client/verta/verta/pipeline/_registered_pipeline.py

```diff
@@ -0,0 +1,224 @@
+    graph: :class:`~verta.pipeline.PipelineGraph`
+        PipelineGraph object containing all possible steps in the Pipline.
+    """
+
+    def __init__(
+        self,
+        registered_model_version: RegisteredModelVersion,
+        graph: PipelineGraph,
+    ):
+        """Create a Pipeline instance from an existing RegisteredModelVersion object
+        and the provided pipeline graph.
+
+        Name and ID are captured once upon creation to avoid additional HTTP calls
+        to refresh the cache of the RMV, because pipelines are immutable.
+        """
+        self._registered_model_version = registered_model_version
+        self._name = self._registered_model_version.name
+        self._id = self._registered_model_version.id
+        self._graph = graph
+        self._graph_steps = self._graph.steps
+        # throws an exception if the graph's steps attr has been inappropriately mutated.
+
+    def __repr__(self):
+        return "\n    ".join(
+            (
+                "RegisteredPipeline:",
+                f"pipeline name: {self.name}",
+                f"pipeline id: {self.id}",
+                f"\n{self._graph}",
+            )
+        )
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def id(self):
+        return self._id
+
+    @property
+    def graph(self):
+        return self._graph
+
+    def copy_graph(self) -> PipelineGraph:
+        """Return a copy of the PipelineGraph object for this pipeline.
+
+        RegisteredPipeline objects are immutable once registered with Verta. This
+        copy can be modified and used to create and register a new RegisteredPipeline.
+
+        Returns
+        -------
+        :class:`~verta.pipeline.PipelineGraph`
+            A deep copy of the PipelineGraph object for this pipeline.
+        """
+        return copy.deepcopy(self._graph)
+
+    def _log_pipeline_definition_artifact(self) -> None:
+        """
+        Log the pipeline definition as an artifact of the registered model version.
+        """
+        with tempfile.NamedTemporaryFile() as temp_file:
+            bytes = json.dumps(self._to_pipeline_definition()).encode("utf-8")
+            temp_file.write(bytes)
+            self._registered_model_version.log_artifact("pipeline.json", temp_file)
+
+    def _to_pipeline_definition(self) -> Dict[str, Any]:
+        """Create a complete pipeline definition dict from a name and PipelineGraph.
+
+        Used in conjunction with the client function for creating a registered
+        pipeline from a pipeline graph.  This gets converted to JSON and uploaded
+        as an artifact to the registered model version for the pipeline by the
+        _log_pipeline_definition_artifact function.
+        """
+        return {
+            "pipeline_version_id": self.id,
+            "graph": self._graph._to_graph_definition(),
+            "predecessors": self._graph._to_steps_definition(),
```

Comment:
I think this should be `"steps"`.

```suggestion
            "steps": self._graph._to_steps_definition(),
```

https://github.com/VertaAI/services/blob/ee2a32d2bfc55d9a82e1f78f55fcb9a69edd278d/deployment/deployment-api/api/schema/definitions/PipelineDefinition.yaml#L12